modules/project_services/outputs.tf

```diff
@@ -23,3 +23,8 @@ output "enabled_apis" {
   description = "Enabled APIs in the project"
   value       = [for api in google_project_service.project_services : api.service]
 }
+
+output "enabled_api_identities" {
+  description = "Enabled API identities in the project"
+  value       = { for i in google_project_service_identity.project_service_identities : i.service => i.email }
```

Comment:
@umairidris can we modify these hardcoded identities to use this now?
https://github.com/terraform-google-modules/terraform-google-project-factory/blob/099cdcce28642b045337b4ca0a0c54a9949d9285/modules/shared_vpc_access/main.tf#L23

Reply:
The challenge with using this resource is you will need to manually grant any roles that were previously automatically granted on activation (e.g. for healthcare SA the healthcare.serviceAgent role is automatically granted).

So if using this resource it should be fine but you will also need to ensure you grant any permissions that you were previously assuming were automatically granted, else it will not have the right permissions to do its job.

Reply:
I think we might want to consider adding functionality for setting IAM roles on SAs to Project Factory, with sensible defaults which match the old defaults.

Reply:
Could you elaboroate?

Currently the input from user looks something like this:

```
activate_api_identities = [{
    api = "healthcare.googleapis.com"
    roles = [
      "roles/healthcare.serviceAgent",
      "roles/bigquery.jobUser",
    ]
  }]
```

Do you want us to add additional helpers to set roles/healthcare.serviceAgent by default? If so, should we manually maintain a mapping of API to default role? I am afraid that might be inconsistent if we miss certain APIs, though https://cloud.google.com/iam/docs/understanding-roles#service-management-roles might help. I'll start an internal thread with the team on potential automation.

Reply:
Got it, I think we can merge as-is and evolve based on the thread.